crates/rpc/src/method/get_storage_proof.rs

```diff
@@ -406,25 +406,29 @@ fn get_contract_proofs(
     };
 
     let storage_root_hash = tx
-        .storage_trie_node_hash(storage_root_idx)
+        .storage_trie_node_hash(storage_root_idx.get())
         .context("Querying storage root hash")?
         .context("Storage root hash missing")?;
 
     let Some(contract_addresses) = contract_addresses.as_ref() else {
         return Ok((storage_root_hash, NodeHashToNodeMappings(vec![]), vec![]));
     };
 
-    let nodes =
-        StorageCommitmentTree::get_proofs(tx, block_number, contract_addresses, storage_root_idx)?
-            .into_iter()
-            .flatten()
-            .map(|(node, node_hash)| NodeHashToNodeMapping {
-                node_hash,
-                node: ProofNode(node),
-            })
-            .collect::<HashSet<_>>()
-            .into_iter()
-            .collect();
+    let nodes = StorageCommitmentTree::get_proofs(
+        tx,
+        block_number,
+        contract_addresses,
+        storage_root_idx.get(),
```

Comment:
This is also suspicious: `StorageCommitmentTree::get_proofs()` is taking a root index as an `u64` here -- we should avoid that.